sky/task.py

```diff
@@ -472,6 +472,8 @@ def update_envs(
         Raises:
           ValueError: if various invalid inputs errors are detected.
         """
+        # NOTE(dev): This should be called before task.set_resources() because
+        # of the docker login config.
```

Comment:
Should we instead, at the end of `set_env` call the `_with_docker_login_config` if we find the docker login config in the new envs, so we don't have to requiring the `set_envs` to be called before `set_resources`?

Reply:
Good point! Done. Thanks!